compiler/next/include/chpl/uast/RealLiteral.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_REALLITERAL_H
+#define CHPL_UAST_REALLITERAL_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/Literal.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents a floating point literal that is not imaginary.
+  That is, it is a "real" number. Examples include ``0.0``, and `3e24`.
+ */
+class RealLiteral final : public Literal {
+ private:
+  double value_;
```

Comment:
I guess we're assuming literal numeric types (RealLiteral, ImagLiteral, etc.)  fit in double? Would we want to support different  precision, arbitrary precision, of fixed-point datatypes at some point?

Reply:
It's just for literals like `1.0` which always have type `real(64)` aka `double`. In Chapel we have `param` values and logic instead of needing literals of different types. So we don't need `1u` but rather `1:uint` is just as good.